src/Tasks/Microsoft.NET.Build.Tasks/LockFileExtensions.cs

```diff
@@ -12,6 +12,29 @@ namespace Microsoft.NET.Build.Tasks
 {
     internal static class LockFileExtensions
     {
+        public static LockFileTarget GetTargetAndThrowIfNotFound(this LockFile lockFile, NuGetFramework framework, string runtime)
+        {
+            LockFileTarget lockFileTarget = lockFile.GetTarget(framework, runtime);
+
+            if (lockFileTarget == null)
+            {
+                string frameworkString = framework.DotNetFrameworkName;
+                string targetMoniker = string.IsNullOrEmpty(runtime) ?
+                    frameworkString :
+                    $"{frameworkString}/{runtime}";
+
+                string message = string.Format(Strings.AssetsFileMissingTarget, lockFile.Path, targetMoniker, framework.GetShortFolderName());
+                if (!string.IsNullOrEmpty(runtime))
+                {
+                    message += " " + string.Format(Strings.AssetsFileMissingRuntimeIdentifier, runtime);
+                }
```

Comment:
We should avoid concatenating messages like this. It makes translation more difficult. It's better to duplicate the fragments to give translators maximum flexibility. I see we have a similar pattern in conflict resolution, but that should be fixed as well.